docs/src/main/tut/datatypes/eval.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: docs
+title:  "Eval"
+section: "data"
+source: "core/src/main/scala/cats/Eval.scala"
+scaladoc: "#cats.Eval"
+---
+# Eval
+
+Eval is a data type for controlling synchronous evaluation.
+Its implementation is designed to provide stack-safety at all times using a technique called trampolining.
+
+There are two different factors that play into evaluation: memoization and laziness.
+
+Memoized evaluation evaluates an expression only once and then remembers (memoizes) that value.
+Lazy evaluation refers to when the expression is evaluated.
+We talk about eager evaluation if the expression is immediately evaluated when defined and about lazy evaluation if the expression is evaluated when it's first used.
+
+For example, in Scala, a `lazy val` is both lazy and memoized, a method definition `def` is lazy, but not memoized, since the body will be evaluated on every call.
+A normal `val` evaluates eagerly and also memoizes the result.
+
+`Eval` is able to express all of these evaluation strategies and allows us to chain computations using its `Monad` instance.
+
+#### Eval.now
+
+First of the strategies is eager evaluation, we can construct an `Eval` eagerly using `Eval.now`:
+
+
+```tut:book
+import cats.Eval
+import cats.implicits._
+
+
+val eager = Eval.now {
+  println("Running expensive calculation...")
+  1 + 2 * 3
+}
+```
+
+
+We can run the computation using the given evaluation strategy anytime by using the `value` method.
+
+```tut:book
+eager.value
+
+```
+
+#### Eval.later
+
+If we want lazy evaluation, we can use `Eval.later`:
```

Comment:
Might be worthy mentioning here that there are two advantages of `Later` over `lazy val` 
1. `Later` allows the thunk being GC'd after evaluation and
2. `Later`'s synchronization lock locks itself while lazy vals lock their enclosing class.